stx-labs/stacks.js
# Fix Linkedin verification proof

**Description**

Fixes blockstack/blockstack.js#550

The DOM parsing selectors for the Linkedin
verification have been updated so that the
cheerio library could locate the correct data

The HTML generated from cheerio does not match
the HTML rendered in the browser which makes
it more difficult to parse for the correct
selectors

**Testing**

1. Prepare a local development instance of blockstack-browser
2. Ensure that you are using this [forked branch](https://github.com/otech47/blockstack.js/tree/test-linkedin-verification) (edit package.json to include `"blockstack": "github:otech47/blockstack.js#test-linkedin-verification"`) of blockstack.js before running `npm install`
3. Authenticate with any Blockstack ID and follow the steps to verify your LinkedIn profile
4. Verify the publicly posted link and confirm the blue checkmark appears

I ran this test manually using an account where I had attempted a LinkedIn verification before and was able to get to step 4 successfully. I don't anticipate any issues with new accounts attempting the same since this is simply a matter of changing selectors.

I also revised the unit tests because it seems a [test case was added ](https://github.com/blockstack/blockstack.js/commit/7704d2710ca6b6dedf82694f67568e15fa17578b#diff-5fffa10b590af6297fc38e7d04821de3R71) to handle the Linkedin verification failure. I removed that test and added extra test data to mock the HTML.

This is my 1st PR so I figured I might as well open this now to see if there is any testing guidance or comments from the maintainers or previous contributors @yknl  @shreyasthiagaraj on anything additional needed here.  For example I wasn't sure whether to include the `dist/blockstack.js` file or not.
